sys/net/network_layer/ng_sixlowpan/frag/rbuf.c

```diff
@@ -166,8 +137,18 @@ void rbuf_add(ng_netif_hdr_t *netif_hdr, ng_sixlowpan_frag_t *frag,
         netif_hdr->if_pid = iface;
         LL_APPEND(entry->pkt, netif);
 
-        DEBUG("6lo rbuf: datagram complete, send to self\n");
-        ng_netapi_receive(thread_getpid(), entry->pkt);
+        if (entry->compressed) {
+            DEBUG("6lo rbuf: datagram complete, send to self for decompression\n");
+            ng_netapi_receive(thread_getpid(), entry->pkt);
+        }
+        else {
+            DEBUG("6lo rbuf: datagram complete, send to IPv6 listeners\n");
+            if (!ng_netapi_dispatch_receive(NG_NETTYPE_IPV6, NG_NETREG_DEMUX_CTX_ALL,
+                                            entry->pkt)) {
+                DEBUG("6lo rbuf: No receivers for this packet found\n");
+                ng_pktbuf_release(entry->pkt);
+            }
+        }
         _rbuf_rem(entry);
```

Comment:
Because it is reassembled and does not need to be stored in the reassembly buffer anymore (because it is forwarded to IPv6 (in l142)/reissued for decompression (in l149))?